app/models/responses.go

```diff
@@ -83,6 +84,9 @@ func (r *ResultMessage) MarshalXML(e *xml.Encoder, start xml.StartElement) error
 	}
 	var messageAttrs []MessageAttributes
 	for key, value := range r.MessageAttributes {
+		if value.DataType == "Binary" {
+			value.BinaryValue = []byte(base64.StdEncoding.EncodeToString(value.BinaryValue))
+		}
```

Comment:
Here is the related fix for Bug 1.

In the AWS query protocol, [blobs are sent and received after being base64 encoded](https://smithy.io/2.0/aws/protocols/aws-query-protocol.html) , so the binary value should be base64 encoded.

Reply:
I don't know about this.  I understand that's part of the query protocol, but that whole thing is deprecated in AWS.  Based on my tests this works, as is, for the existing sdks I've tried.  What exactly isn't working for you?

Reply:
Yep, as you said this part is for the query protocol. The latest version of AWS SDK for Java also works without this part.

I thought that future versions of GOAWS still keep support the query protocol, so I added it.
Sorry I don't remember clearly but, Is the policy that "users who need to use aws-query should use older versions of goaws"? (It also makes sense)

Reply:
Well, no, that's not the policy per say.  I hadn't given it that much thought if I'm honest.  But if you think about it like this - this is a tool for folks to run their own apps against instead AWS directly.  So my main goal is to make it work in a performant/lightweight way and replicate as much of the AWS SQS/SNS "contract" as I can.  

So with that in mind - we don't NEED to match what AWS does, as long as we respond the same way that it would.  We just need to match their contracts right?  At any rate we can't know what they're doing all the times as their docs are generally garbage in my opinion.  

So, I don't care about things like security, and base64 encoding much, because, it's really up to the user here to worry about their input and output - because it's an app running where a user puts it.  "Do what you will with it." so to speak, as long as an SDK doesn't fail on it - "garbage in garbage" out.  

**Here's what I need:**
So all that said - this change is doing a couple of things.  I need to know why we need them.  The way I had it lately was working for me, based on the contract I had researched.  What I'd like to see is what you're failing with, how you're recreating it, and why we think these changes will fix them.  

Sorry - this is a lot for me to juggle long term and I need a little grooming/refining (in agile terms), before I let it in.  Otherwise I'll be adrift if someone else complains about it or something similar later.

As always thank you for giving it so much work and thought.  I really appreciate it and without you I wouldn't be able to keep this tool as current and viable as it is.